core/src/main/java/org/apache/calcite/sql/fun/SqlLibrary.java

```diff
@@ -172,5 +187,20 @@ private static void addParent(Set<SqlLibrary> set, SqlLibrary library) {
       builder.put(value.fun, value);
     }
     MAP = builder.build();
+    final Map<SqlLibrary, ImmutableSet.Builder<SqlLibrary>> childrenMap = new HashMap<>();
```

Comment:
A simpler - slightly less efficient - implementation would use a function
```
  Set<SqlLibrary> children(SqlLibrary library) {
    return Stream.of(values()).filter(lib -> lib.parent == library).collect(toSet());
  }
```
and then you could build the map directly (without an intermediate map of `ImmutableSet.builder`)

Reply:
Updated as you suggested.